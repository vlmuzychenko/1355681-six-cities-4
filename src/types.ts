export interface CityInterface {
  name: string,
  coords: number[],
}

export interface OfferInterface {
  id: number,
  title: string,
  price: number,
  previewImage: string,
  images: string[],
  description: string,
  rating: number,
  isPremium: boolean,
  isFavorite: boolean,
  type: string,
  bedrooms: number,
  maxAdults: number,
  goods: string[],
  host: {
    name: string,
    super: boolean,
    avatarUrl: string,
  },
  city: CityInterface,
  coords: number[],
}

export interface ReviewInterface {
  id: number,
  user: {
    name: string,
    avatarUrl: string,
    id: number,
    isPro: boolean,
  },
  rating: number,
  comment: string,
  date: string,
}

export interface AuthorizationInfoInterface {
  avatarUrl: string,
  email: string,
  id: number,
  isPro: boolean,
  name: string,
}
