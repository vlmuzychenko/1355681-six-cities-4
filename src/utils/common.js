export const getRatingInPercent = (rating) => {
  return `${Math.round(rating) * 100 / 5}%`;
};

export const getPluralizedString = (count, noun, suffix = `s`) => {
  return `${noun}${count !== 1 ? suffix : ``}`;
};

export const getItemsById = (items, id) => {
  return items.filter((item) => item.id === id);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getOffersByCity = (offers, city) => offers.filter((offer) => {
  return offer.city.name === city.name;
});
