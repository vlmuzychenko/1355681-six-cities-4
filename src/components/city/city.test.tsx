import React from "react";
import renderer from "react-test-renderer";
import City from "./city";
import {CityInterface} from "../../types";
import {noop} from "../../utils/common";

const cityMock: CityInterface = {
  name: `Paris`,
  coords: [48.865, 2.35],
  zoom: 12,
};

const isActiveTrueMock = true;
const isActiveFalseMock = false;

it(`Should Active City render correctly`, () => {
  const tree = renderer
    .create(
        <City
          city={cityMock}
          isActive={isActiveTrueMock}
          onCityNameClick={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should City render correctly`, () => {
  const tree = renderer
    .create(
        <City
          city={cityMock}
          isActive={isActiveFalseMock}
          onCityNameClick={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});


