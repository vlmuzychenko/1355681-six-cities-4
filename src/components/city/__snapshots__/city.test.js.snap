// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Active City render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;

exports[`Should City render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item false"
    href="#"
    onClick={[Function]}
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;
